spec/requests/api/v1/moves_controller_index_spec.rb

```diff
@@ -32,14 +33,19 @@
           }
         end
         let(:params) { { filter: filters } }
+        let(:ability) { Ability.new }
+
+        before do
+          allow(Ability).to receive(:new).and_return(ability)
+        end
 
         it 'delegates the query execution to Moves::Finder with the correct filters', skip_before: true do
           moves_finder = instance_double('Moves::Finder', call: Move.all)
           allow(Moves::Finder).to receive(:new).and_return(moves_finder)
 
           get '/api/v1/moves', headers: headers, params: params
 
-          expect(Moves::Finder).to have_received(:new).with(from_location_id: from_location_id)
+          expect(Moves::Finder).to have_received(:new).with({ from_location_id: from_location_id }, ability)
```

Comment:
I know you haven't introduced this mocking (🤷‍♂) , but would this be a good opportunity to remove it? I'd expect our functional tests to build test data and assert the behaviour, rather than checking a particular method has been called somewhere. This doesn't block this one, but maybe a candidate for another PR?

Reply:
I can do it, but I think that being a rather big change in direction we must think about duplication.
Finders are actually "in charge" of retrieving the data, and are tested. I have nothing against removing the mocks but we'll most probably end up with a lot of overlapping tests on the same functionalities.
IMVHO it makes sense to have request specs mock the finders and test how a request to the controller turns into a call to the service object and then test the service object in isolation.

Reply:
Happy to leave this for this PR. I'd assumed we use the request specs as a functional end-to-end test, but if that's not the case we'll want to think about how we introduce the latter. Maybe something to discuss with @starswan and Martyn once he's settled in next week?